packages/use-intl/src/core/createFormatter.tsx

```diff
@@ -158,6 +158,40 @@ export default function createFormatter({
     );
   }
 
+  function dateTimeRange(
+    /** If a number is supplied, this is interpreted as a UTC timestamp. */
+    start: Date | number,
+    /** If a number is supplied, this is interpreted as a UTC timestamp. */
+    end: Date | number,
+    /** If a time zone is supplied, the `value` is converted to that time zone.
+     * Otherwise the user time zone will be used. */
+    formatOrOptions?: string | DateTimeFormatOptions
+  ) {
+    return getFormattedValue(
+      [start, end],
+      formatOrOptions,
+      formats?.dateTime,
+      (options) => {
+        if (!options?.timeZone) {
+          if (globalTimeZone) {
+            options = {...options, timeZone: globalTimeZone};
+          } else {
+            onError(
+              new IntlError(
+                IntlErrorCode.ENVIRONMENT_FALLBACK,
+                process.env.NODE_ENV !== 'production'
+                  ? `The \`timeZone\` parameter wasn't provided and there is no global default configured. Consider adding a global default to avoid markup mismatches caused by environment differences. Learn more: https://next-intl-docs.vercel.app/docs/configuration#time-zone`
+                  : undefined
+              )
+            );
+          }
+        }
```

Comment:
Maybe we can share this part with `dateTime` to reduce bundle size.

```suggestion
        if (!options?.timeZone) {
          options = applyGlobalTimeZone(options);
        }
```

… with:

```tsx
  function applyGlobalTimeZone(options?: DateTimeFormatOptions) {
    if (globalTimeZone) {
      options = {...options, timeZone: globalTimeZone};
    } else {
      onError(
        new IntlError(
          IntlErrorCode.ENVIRONMENT_FALLBACK,
          process.env.NODE_ENV !== 'production'
            ? `The \`timeZone\` parameter wasn't provided and there is no global default configured. Consider adding a global default to avoid markup mismatches caused by environment differences. Learn more: https://next-intl-docs.vercel.app/docs/configuration#time-zone`
            : undefined
        )
      );
    }
    return options;
  }
```

Reply:
yeah sure!